### PR TITLE
Added NPM publish workflow

### PR DESCRIPTION
**Pull request by Razican**
This PR adds a release workflow to publish NPM packages on each release in an autonomous way.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -40,6 +40,42 @@ jobs:
             --no-git-commit \
             skip
 
+  npm_publish:
+    name: Publish crates
```
Comment (nekevss) [inline .github/workflows/release.yml:44]: thought: Should the name of job be changed to: Publish boa_wasm.

Comment (Razican) [reply .github/workflows/release.yml]: Fixed :)

**Review by HalidOdat — APPROVED** [review]
Nice! looks good to me! :)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -40,6 +40,42 @@ jobs:
...
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: wasm32-unknown-unknown
+          profile: minimal
+          override: true
```
Comment (jedel1043) [inline .github/workflows/release.yml:57]: Would prefer if we used Dtolnay's action, since actions-rs repos are unmaintained.

**Review by jedel1043 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (Razican) [reply .github/workflows/release.yml:57]: I just noticed we are using the actions-rs actions everywhere in our codebase. Should we do this change separately, to migrate to the dtolnay's action?

Comment (jedel1043) [reply .github/workflows/release.yml:57]: Yep, that would be preferable.

**Review by jedel1043 — APPROVED** [review]
This will need a follow up PR to remove all actions from `actions-rs` from the codebase.

**Pull request merged**